builtin/providers/cloudstack/resource_cloudstack_network.go

```diff
@@ -144,6 +146,8 @@ func resourceCloudStackNetworkCreate(d *schema.ResourceData, meta interface{}) e
 
 	d.SetId(r.Id)
 
+	setTags(cs, d, "network")
```

Comment:
You ignore the error value returned here... I think we should check the error and return the error if not nil before moving on to the `resourceCloudStackNetworkRead` call.